$MirOS: ports/security/pgp+/patches/patch-src_pgp_c,v 1.1.7.1 2005/03/18 15:51:41 tg Exp $
--- src/pgp.c.orig	Fri Mar 23 10:59:25 2001
+++ src/pgp.c	Thu Mar 22 00:37:37 2007
@@ -130,6 +130,7 @@
 #ifdef UNIX
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <fcntl.h>
 #endif
 
 #include "system.h"
@@ -954,6 +955,23 @@ in AUTOEXEC.BAT file.\n"));
 
     initsigs();			/* Catch signals */
     noise();			/* Start random number generation */
+
+#ifdef	UNIX
+    /* seed with OS random number generator, too */
+    {
+	char urand[1024];
+	int f;
+
+	f = open("/dev/arandom", O_RDONLY, 0);
+	if (f == -1)
+		f = open("/dev/urandom", O_RDONLY, 0);
+	if (f != -1) {
+		randPoolAddBytes(urand,
+		    read(f, urand, 1024));
+		close(f);
+	}
+    }
+#endif
 
     if (keyflag) {
 	status = do_keyopt(keychar);
